presto-password-authenticators/src/main/java/io/prestosql/plugin/password/filebasedauthenticator/FileBasedAuthenticator.java

```diff
@@ -0,0 +1,76 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.password.filebasedauthenticator;
+
+import com.google.common.cache.Cache;
+import com.google.common.cache.CacheBuilder;
+import io.airlift.log.Logger;
+import io.prestosql.spi.PrestoException;
+import io.prestosql.spi.StandardErrorCode;
+import io.prestosql.spi.security.AccessDeniedException;
+import io.prestosql.spi.security.BasicPrincipal;
+import io.prestosql.spi.security.PasswordAuthenticator;
+
+import javax.inject.Inject;
+
+import java.io.IOException;
+import java.security.Principal;
+import java.util.Map;
+
+import static io.prestosql.plugin.password.filebasedauthenticator.Users.readUsers;
+import static java.lang.String.format;
+
+public class FileBasedAuthenticator
+        implements PasswordAuthenticator
+{
+    private static final Logger log = Logger.get(FileBasedAuthenticator.class);
+    private long maximumSize = 1000; //maximum size of the LRU cache for storing auth token
+    private Cache<User.PasswordStore, Boolean> bcryptCache;
+    private Cache<User.PasswordStore, Boolean> pbkdf2Cache;
+    private Map<String, User> authMap;
+
+    @Inject
+    public FileBasedAuthenticator(FileBasedAuthenticationConfig config) throws IOException
```

Comment:
It looks like `IOException` is not thrown by this method

Reply:
Done